jsapi/src/main.js

```diff
@@ -6,28 +6,75 @@ goog.require('ol.events');
 goog.require('ol.Map');
 goog.require('ol.Overlay');
 goog.require('ol.View');
+goog.require('ol.layer.Image');
 goog.require('ol.layer.Tile');
 goog.require('ol.source.OSM');
+goog.require('ol.source.ImageWMS');
+goog.require('ol.source.WMTSRequestEncoding');
 
 /**
  * @classdesc
  * The map is the core component of the Geoportail V3 API.
  *
  * @constructor
  * @extends {ol.Map}
- * @param {olx.MapOptions} options Map options.
+ * @param {luxx.MapOptions} options Map options.
  * @export
  */
 lux.Map = function(options) {
-  options.view = new ol.View({
-    zoom: 5,
-    center: [0, 0]
-  });
-  options.layers = [new ol.layer.Tile({
-    source: new ol.source.OSM()
-  })];
+
+  var layers    = [];
+  var defaultBg = 'basemap_2015_global';
+
+  /**
+   * @private
+   * @type {Object} config A JSON layers config object.
+   */
+  this.layersConfig = null;
+
+  this.promise = null;
```

Comment:
I'm not sure if it is required or not but `promise` could be documented as `@private` as well.